viewing_party/party.py

```diff
@@ -1,23 +1,197 @@
+
+def get_most_watched_genre(user_data):
+    watched = user_data["watched"]
+
+    if not watched:
+        return None
+    
+    genre_counts = {}
+    for movie in watched:
+        genre = movie.get("genre")
+        if genre not in genre_counts:
+            genre_counts[genre] = 1
+        else:
+            genre_counts[genre] += 1
+
+    # most_watched_genre = max(genre_counts, key=genre_counts.get)
+
+    max_count = 0
+    
+    for genre, count in  genre_counts.items():
+
+        if count > max_count:
+
+            max_count = count
+            max_watched_genre = genre
+
+
+    return max_watched_genre
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    if not user_data:
+        return None
+    
+    user_watched_movies = user_data["watched"]
+    friends_watched_movies = []  
+
+    for friend in user_data["friends"]:
+        friends_watched_movies.extend(friend["watched"])
+
+    unique_watched = []
+
+    for movie in user_watched_movies:
+        if movie not in friends_watched_movies:
+            unique_watched.append(movie)
+
+    return unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    if not user_data:
+        return None
+    
+    user_watched_movies = user_data["watched"]
+    friends_watched_movies = []  
+
+    for friend in user_data["friends"]:
+        friends_watched_movies.extend(friend["watched"])
+
+    unique_watched = []
+
+    for movie in friends_watched_movies:
+        if movie not in user_watched_movies and movie not in unique_watched:
+            unique_watched.append(movie)
+    
+    return unique_watched
+
+
+
+
+
+
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    
+    user_watched = user_data["watched"]
+    subscriptions = user_data["subscriptions"]
+    
+    friend_watched =[]
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friend_watched:
+                friend_watched.append(movie)
+                
+    recs = []
+    for movie in friend_watched:
+       
+        if movie in user_watched:
+            continue
+
+       
+        if movie["host"] not in subscriptions:
+            continue
+
+       
+        recs.append(movie)
+    return recs
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    most_watched_genre = get_most_watched_genre(user_data)
+
+    if most_watched_genre is None:
+        return []
+    
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched:
+                friends_watched.append(movie)

```

Comment:
As mentioned before, this looks like we are making a list of all the unique movies our friends have watched. I wonder if we can use a function we already built to make this a little easier?